pkg/webhook/controlplane/ensurer.go

```diff
@@ -149,6 +150,154 @@ func keepNodeNetworkEnvVarIfPresentInOldDeployment(new, old *appsv1.Deployment,
 	}
 }
 
+// EnsureAdditionalProvisionUnits ensures that additional required system units are added, that are required during provisioning.
+func (e *ensurer) EnsureAdditionalProvisionUnits(ctx context.Context, gctx gcontext.GardenContext, new, _ *[]extensionsv1alpha1.Unit) error {
+
+	cluster, err := gctx.GetCluster(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Define LVM setup unit
+	var lvmSetup = `
+[Unit]
+        Description=LVM Setup
+        ConditionFirstBoot=yes
+        DefaultDependencies=no
+        Before=local-fs-pre.target
+        [Service]
+        Type=oneshot
+        Restart=on-failure
+        RemainAfterExit=yes
+        ExecStart=/opt/bin/lvm.sh
+        [Install]
+        WantedBy=multi-user.target
+`
+	// Define LVM mount unit
+	var lvmMountContainerd = `
+[Unit]
+        Description=Mount LVM to containerd dir
+        After=lvm-setup.service
+        Before=containerd.service
+        [Mount]
+        What=/dev/vg-containerd/vol_containerd
+        Where=/var/lib/containerd
+        Type=ext4
+        Options=defaults
+        [Install]
+        WantedBy=local-fs.target
+`
+	operatingsystems := make(map[string]int)
+	for _, worker := range cluster.Shoot.Spec.Provider.Workers {
+		if worker.DataVolumes != nil {
+			// fail if multiple OSes are used
+			operatingsystems[worker.Machine.Image.Name]++
+			if len(operatingsystems) > 1 {
+				return fmt.Errorf("multiple operatingsystems used")
```

Comment:
we should also specify that only one homogenous is supported in the error

Reply:
m,aybe also include the name. Otherwiese the `Errorf` is not needed and we can use `errors.New`